Tasks/AzureRmWebAppDeployment/azurermwebappdeployment.ts

```diff
@@ -83,44 +103,46 @@ async function run() {
+                var isFolderBasedDeployment = deployUtility.isInputPkgIsFolder(webDeployPkg);
+                var applyFileTransformFlag = JSONFiles.length != 0 || xmlTransformation || xmlVariableSubstitution;
 
-            if (applyFileTransformFlag || generateWebConfig) {
-                var folderPath = await deployUtility.generateTemporaryFolderForDeployment(isFolderBasedDeployment, webDeployPkg);
+                if (applyFileTransformFlag || generateWebConfig) {
+                    var folderPath = await deployUtility.generateTemporaryFolderForDeployment(isFolderBasedDeployment, webDeployPkg);
 
-                if (generateWebConfig) {
-                    tl.debug('parsing web.config parameters');
-                    var webConfigParameters = ParameterParser.parse(webConfigParametersStr);
-                    generateWebConfigUtil.addWebConfigFile(folderPath, webConfigParameters, virtualApplicationPhysicalPath);
-                }
-                if (applyFileTransformFlag) {
-                    var isMSBuildPackage = !isFolderBasedDeployment  && (await deployUtility.isMSDeployPackage(webDeployPkg));
-                    fileTransformationsUtility.fileTransformations(isFolderBasedDeployment, JSONFiles, xmlTransformation, xmlVariableSubstitution, folderPath, isMSBuildPackage);
-                }
+                    if (generateWebConfig) {
+                        tl.debug('parsing web.config parameters');
+                        var webConfigParameters = ParameterParser.parse(webConfigParametersStr);
+                        generateWebConfigUtil.addWebConfigFile(folderPath, webConfigParameters, virtualApplicationPhysicalPath);
+                    }
+                    if (applyFileTransformFlag) {
+                        var isMSBuildPackage = !isFolderBasedDeployment && (await deployUtility.isMSDeployPackage(webDeployPkg));
+                        fileTransformationsUtility.fileTransformations(isFolderBasedDeployment, JSONFiles, xmlTransformation, xmlVariableSubstitution, folderPath, isMSBuildPackage);
+                    }
 
-                var output = await deployUtility.archiveFolderForDeployment(isFolderBasedDeployment, folderPath);
-                tempPackagePath = output.tempPackagePath;
-                webDeployPkg = output.webDeployPkg;
-            }
+                    var output = await deployUtility.archiveFolderForDeployment(isFolderBasedDeployment, folderPath);
+                    tempPackagePath = output.tempPackagePath;
+                    webDeployPkg = output.webDeployPkg;
+                }
 
-            if(virtualApplication) {
-                publishingProfile.destinationAppUrl += "/" + virtualApplication;
+                if (virtualApplication) {
+                    publishingProfile.destinationAppUrl += "/" + virtualApplication;
+                }
             }
-
             if(webAppUri) {
```

Comment:
repeat: 2

Reply:
The value may change in case of Windows webapp.